src/tests/tables.test.js

```diff
@@ -3,20 +3,92 @@ const glob = require("glob");
 
 describe("duplicate indices", () => {
   it("should contain unique indices", () => {
+    let errors = [];
+    const fileIndices = {};
+
+    forEachFile((filename, entry) => {
+      if (filename in fileIndices === false) {
+        fileIndices[filename] = new Set();
+      }
+
+      if (fileIndices[filename].has(entry.index)) {
+        errors.push(`${filename}: Index '${entry.index}' already exists.`);
+      }
+
+      fileIndices[filename].add(entry.index);
+    });
+
+    expect(errors).toEqual([]);
+  });
+});
+
+describe("api references", () => {
+  it("should not contain broken links", () => {
     let errors = [];
     let files = glob.sync("src/*.json");
 
-    for (const file of files) {
-      const fileText = fs.readFileSync(file, "utf8");
-      const fileJSON = JSON.parse(fileText);
-      const indices = new Set();
-      fileJSON.forEach((entry) => {
-        if (indices.has(entry.index)) {
-          errors.push(`${file}: Index '${entry.index}' already exists.`);
+    let resources = {};
+
+    forEachFile((filename, entry) => {
+      if (entry.url === undefined) return;
+
+      if (entry.index === undefined) {
+        errors.push(`${filename}: Entry with URL '${entry.url}' should have an index.`);
+      }
+
+      resources[entry.url] = { index: entry.index, name: entry.name };
+    });
+
+    forEachFile((filename, topLevelEntry) => {
+      recurseIntoObject(topLevelEntry, (subEntry) => {
+        if (!subEntry.hasOwnProperty("url")) return;
+
+        if (resources[subEntry.url] === undefined) {
+          errors.push(`${filename}: URL '${subEntry.url}' not found.`);
+        } else {
+          if (resources[subEntry.url].name !== undefined && resources[subEntry.url].name !== subEntry.name) {
+            errors.push(`${filename}: Name mismatch for reference to '${subEntry.url}', '${subEntry.name}' should be '${resources[subEntry.url].name}'`);
+          }
+
+          if (subEntry.index !== undefined && resources[subEntry.url].index !== subEntry.index) {
+            errors.push(`${filename}: Index mismatch for reference to '${subEntry.url}', '${subEntry.index}' should be '${resources[subEntry.url].index}'`);
+          }
         }
-        indices.add(entry.index);
       });
-    }
+    });
+
     expect(errors).toEqual([]);
   });
 });
+
+/**
+ * Calls the callback for top-level objects/arrays in all JSON files.
+ *
+ * @param (function(string, object)) callback Called with filename and each
+ *     top-level entry.
+ */
+const forEachFile = (callback) => {
```

Comment:
Would this be better named as `forEachEntryInFile`?

Reply:
Makes sense, but I went with `forEachFileEntry()` instead because it iterates over all files.